atom-manager-play-lib/src/main/scala/com.gu.atom/play/AtomAPIActions.scala

```diff
@@ -15,30 +15,41 @@ trait AtomAPIActions extends Controller {
 
   val livePublisher: LiveAtomPublisher
   val previewPublisher: PreviewAtomPublisher
-  val dataStore: DataStore
+  val previewDataStore: PreviewDataStore
+  val publishedDataStore: PublishedDataStore
 
   private def jsonError(msg: String): JsObject = JsObject(Seq("error" -> JsString(msg)))
 
   def publishAtom(atomId: String) = Action { implicit req =>
-    dataStore.getAtom(atomId) match {
-      case Some(atom) =>
+
+    val revisionNumber = publishedDataStore.getAtom(atomId) match {
+      case Some(atom) => atom.contentChangeDetails.revision + 1
+      case None => 1
+    }
+
+    previewDataStore.getAtom(atomId) match {
+      case Some(atom) => {
         val updatedAtom = atom.copy(
-          contentChangeDetails = atom.contentChangeDetails.copy(
-            published = Some(ChangeRecord((new Date()).getTime(), None))
+          contentChangeDetails = atom.contentChangeDetails.copy(published = Some(ChangeRecord((new Date()).getTime(), None)))
+        ).withRevision(revisionNumber)
+
+        saveAtom(updatedAtom)
+      }
+      case None => NotFound(jsonError(s"No such atom $atomId"))
+    }
+  }
+
+  private def saveAtom(updatedAtom: Atom) = {
```

Comment:
could we change the name of this method to something like `savePublishedAtom`, or something more descriptive? Especially as it actually does the publishing as well as saving it.